src/main/java/org/broadinstitute/consent/http/service/MatchService.java

```diff
@@ -142,17 +144,17 @@ public List<Match> createMatchesForConsent(String consentId) {
             List<DataAccessRequest> dars = findRelatedDars(List.of(d.getDataSetId()));
             dars.forEach(dar -> {
                 try {
-                    matches.add(singleEntitiesMatchV2(d, dar));
+                    matches.add(singleEntitiesMatchV3(d, dar));
                 } catch (Exception e) {
-                    logWarn("Error finding  matches for consent: " + consentId);
-                    matches.add(new Match(consentId, dar.getReferenceId(), true, false, MatchAlgorithm.V2, List.of()));
+                    logWarn("Error finding matches for consent: " + consentId);
+                    matches.add(matchFailure(consentId, dar.getReferenceId(), List.of()));
                 }
             });
         });
         return matches;
     }
 
-    private Match singleEntitiesMatchV2(Dataset dataset, DataAccessRequest dar) {
+    private Match singleEntitiesMatchV3(Dataset dataset, DataAccessRequest dar) {
```

Comment:
nit: should we add a test for this function or to `matchSuccess` to ensure that the logic for abstain/approve/deny is mapping to the right booleans?

Reply:
I added tests to have complete coverage of `singleEntitiesMatchV3`.